src/app/api/activity-logs/services/activity-log.service.ts

```diff
@@ -0,0 +1,143 @@
+import { BaseService } from '@api/core/services/base.service'
+import User from '@api/core/models/User.model'
+import { z } from 'zod'
+import { ActivityType, AssigneeType, Comment } from '@prisma/client'
+import { DBActivityLogArraySchema, DBActivityLogDetails, SchemaByActivityType } from '@api/activity-logs/const'
+import { CopilotAPI } from '@/utils/CopilotAPI'
+import { InternalUsers } from '@/types/common'
+import { CommentService } from '../../comment/comment.service'
+import { UserRole } from '../../core/types/user'
+
+export class ActivityLogService extends BaseService {
+  constructor(user: User) {
+    super(user)
+  }
+
+  async get(taskId: string) {
+    const activityLogs = await this.db.$queryRaw`
+        select "ActivityLogs".*
+        from "ActivityLogs"
+                 left join public."Comments" C
+                           on "ActivityLogs"."taskId" = C."taskId" AND ("ActivityLogs".details::JSON ->> 'id')::text = C.id::text
+        where "ActivityLogs"."taskId" = ${z.string().uuid().parse(taskId)}::uuid
+          AND (
+            "ActivityLogs".type <> ${ActivityType.COMMENT_ADDED}::"ActivityType"
+            OR
+            (
+                "ActivityLogs".type = ${ActivityType.COMMENT_ADDED}::"ActivityType" 
+                AND C."deletedAt" IS NULL AND C."parentId" IS NULL
+            )
+          )
+        ORDER BY "ActivityLogs"."createdAt";
+    `
+    const parsedActivityLogs = DBActivityLogArraySchema.parse(activityLogs)
+
+    const copilotService = new CopilotAPI(this.user.token)
+
+    const promises_getCopilotUsers = parsedActivityLogs.map(async (activityLog) => {
+      if (activityLog.userRole === AssigneeType.internalUser) {
+        return copilotService.getInternalUser(activityLog.userId)
```

Comment:
instead of fetching individual users and clients, can we fetch all the users and clients and filter later?

const internalUsers = copilotService.getInternalUsers();
const clientUsers = copilotService.getClientUsers();

then

inside map, find based on the id?

Reply:
changes made in the commit below